pyanaconda/input_checking.py

```diff
@@ -435,6 +436,27 @@ def run(self, check_request):
         self.result.length_ok = length_ok  # pylint: disable=attribute-defined-outside-init
 
 
+class PasswordFIPSCheck(InputCheck):
+    """Check if the password is valid under FIPS, if applicable."""
+
+    def run(self, check_request):
+        fips_enabled = kernel_arguments.is_enabled("fips")
+
+        if not fips_enabled:
+            self.result.success = True
+            self.result.error_message = ""
+            return
+
+        if len(check_request.password) < constants.FIPS_PASSPHRASE_MIN_LENGTH:
+            self.result.success = False
+            self.result.error_message = _(
+                "In FIPS mode, the passphrase must be at least {} letters long."
+            ).format(constants.FIPS_PASSPHRASE_MIN_LENGTH)
+        else:
+            self.result.success = True
+            self.result.error_message = ""
+
+
```

Comment:
I'd maybe organize the code here a bit differently but it is not important.

Reply:
Changed.

Reply:
Something similar, thank you:)